tests/unit_tests/ensemble_evaluator/test_ensemble_evaluator.py

```diff
@@ -18,11 +12,114 @@
     JOB_STATE_RUNNING,
 )
 
-from .ensemble_evaluator_utils import AutorunTestEnsemble, send_dispatch_event
+from .ensemble_evaluator_utils import send_dispatch_event
+
+
+def test_new_monitor_can_pick_up_where_we_left_off(evaluator):
+    evaluator._start_running()
+    token = evaluator._config.token
+    cert = evaluator._config.cert
+    url = evaluator._config.url
+
+    config_info = evaluator._config.get_connection_info()
+    with Monitor(config_info) as monitor:
+        events = monitor.track()
+        # first snapshot before any event occurs
+        snapshot_event = next(events)
+        snapshot = Snapshot(snapshot_event.data)
+        assert snapshot.status == ENSEMBLE_STATE_UNKNOWN
+        # two dispatchers connect
+        with Client(
+            url + "/dispatch",
+            cert=cert,
+            token=token,
+            max_retries=1,
+            timeout_multiplier=1,
+        ) as dispatch1, Client(
+            url + "/dispatch",
+            cert=cert,
+            token=token,
+            max_retries=1,
+            timeout_multiplier=1,
+        ) as dispatch2:
+            # first dispatcher informs that job 0 is running
+            send_dispatch_event(
+                dispatch1,
+                identifiers.EVTYPE_FM_JOB_RUNNING,
+                f"/ert/ensemble/{evaluator.ensemble.id_}/real/0/step/0/job/0",
+                "event1",
+                {"current_memory_usage": 1000},
+            )
+
+            # second dispatcher informs that job 0 is running
+            send_dispatch_event(
+                dispatch2,
+                identifiers.EVTYPE_FM_JOB_RUNNING,
+                f"/ert/ensemble/{evaluator.ensemble.id_}/real/1/step/0/job/0",
+                "event1",
+                {"current_memory_usage": 1000},
+            )
+            send_dispatch_event(
+                dispatch2,
+                identifiers.EVTYPE_FM_JOB_RUNNING,
+                f"/ert/ensemble/{evaluator.ensemble.id_}/real/1/step/0/job/1",
+                "event1",
+                {"current_memory_usage": 1000},
+            )
+
+        evt = next(events)
+        snapshot = Snapshot(evt.data)
+        assert snapshot.get_job("0", "0", "0").status == JOB_STATE_RUNNING
+        assert snapshot.get_job("1", "0", "0").status == JOB_STATE_RUNNING
+        assert snapshot.get_job("1", "0", "1").status == JOB_STATE_RUNNING
+        # take down first monitor by leaving context
+
+    with Client(
+        url + "/dispatch",
+        cert=cert,
+        token=token,
+        max_retries=1,
+        timeout_multiplier=1,
+    ) as dispatch2:
+        # second dispatcher informs that job 0 is done
```

Comment:
honestly i didn't care much about / read the comments, they're copypasta :grimacing: 
these are actually job runners, yes.
i suppose they were called dispatchers because they connect to the `/dispatch` endpoint of the server.

do you want me to rename them?

Reply:
More the comment bellow :arrow_down:

Reply:
I guess it will be helpful to make the comments right for the next iteration of developers to find their bearings :)

Reply:
comments should be fixed by eb1b445df37b665e44acdc16af14d010950a7cfc

renaming task added to refactor issue